tests/jsco4180.py

```diff
@@ -79,21 +79,33 @@ def test_GIVEN_wrong_component_on_device_WHEN_send_new_sp_THEN_state_machine_abo
 
         # Setting an incorrect component on the device will result in the state machine attempting
         # to rerun the pump and reset components after a delay
-        delay = self.ca.get_pv_value("ERROR:DELAY") # delay before state machine reset
-        self._lewis.backdoor_set_on_device("component_A", value-5)
-        self.ca.assert_that_pv_is("COMP:A", value-5, timeout=5)
-        sleep(delay / 2.0)
-
-        # however if we change setpoint, the loop should start again
-        self.ca.set_pv_value("COMP:A:SP", value-10)
-        self.ca.set_pv_value("COMP:B:SP", value+10)
-        sleep(delay / 2.0 + 1.0) # reset should not have happened
-        self.ca.assert_that_pv_is("COMP:A", value-5)
-        sleep(delay / 2.0) # reset should now have happened
-        self.ca.assert_that_pv_is("COMP:A", value-10, timeout=15)
+        initial_delay = self.ca.get_pv_value("ERROR:DELAY")  # delay before state machine reset
+        delay = 30  # Increase delay to avoid race conditions
+        self.ca.set_pv_value("ERROR:DELAY", delay)
+        try:
+            with self.ca.assert_pv_not_processed("RESET:SP", timeout=1):
+                self._lewis.backdoor_set_on_device("component_A", value - 5)
+                self.ca.assert_that_pv_is("COMP:A", value - 5, timeout=5)
+                sleep(delay / 2.0)
+
+                # however if we change setpoint, the loop should start again
+                self._lewis.backdoor_set_on_device("component_A", value - 5)
```

Comment:
Does this line do anything? `component_A` should already equal `value - 5` from above?

Reply:
No I think you're right it can be removed if the test still passes with it removed